CONTRIBUTING.md

```diff
@@ -74,29 +77,122 @@ make unit_testing
 
 ### Extension integration tests
 
-These tests that package integrations, such as `sklearn` or `openai`
-work properly.
+We use specific use cases to test the entire system.
 
 ```shell
-make ext_testing
+make usecase_testing
 ```
 
-### Databackend integration tests
+### Plugin tests
 
-These tests check that data-backend integrations such as MongoDB or SQLite 
-work as expected.
+We maintain a set of plugins that are tested independently.. If you change the plugin code, you can run the tests for that plugin.
 
 ```shell
-make databackend_testing
+export PYTHONPATH=./
+# Install the plugin you want to test
+pip install -e 'plugins/<PLUGIN_NAME>[test]'
+# Run the tests
+pytest plugins/<PLUGIN_NAME>/plugin_test
 ```
 
-### Smoke tests of cluster mode
 
-These tests check that cluster mode works as expected (`ray`, `vector-search`, `cdc`, `rest`):
+## Lint and type-check the code
 
-```shell
-make smoke_testing
+We use `black` for code formatting, `run` for linting, and `mypy` for type-checking.
+
+You can run the following commands to check the code:
+
+```
+make lint-and-type-check
+```
+
+If you want to format the code, you can run the following command:
 ```
+make fix-and-check
+```
+
+## Contributing to new plugins and templates
+
+The `superduper` project is designed to be extensible and customizable. You can create new plugins and templates to extend the functionality of the project.
+
+### Create a new plugin
+
+Plugins are Python packages that extend the functionality of the SuperDuper project.
+More details about the plugins can be found in the documentation:
+
+- [Data plugins](https://docs.superduper.io/docs/data_plugins/intro)
+- [AI plugins](https://docs.superduper.io/docs/category/ai-plugins)
+
+If you want to create a new plugin, you can follow the steps below:
+
+1. Copy `plugins/template` to `plugins/<PLUGIN_NAME>`
+2. Modify the name of the plugin in the following names:
+   - Plugin name in `pyproject.toml`
+   - Package name `superduper_<PLUGIN_NAME>`
+3. Write the code for the plugin
+4. Create the tests directory `plugin_test/` and write the tests
```

Comment:
Should we say something about which version number to assign to the plugin and how the releases work?